### DISCUSSION TITLE
Setting Working Directory with Context Fails to Find File

### DISCUSSION THREAD

**Discussion by andar1an** [community/community#38511] [Actions] [answered]
I am trying to set my working directory with an environment variable so that I can handle 2 different situations, however, if I use context with the working-directory it does not find files in the directory, but if I hardcode the path, it does. I have tried many different syntax iterations but will paste one iteration below so it is easier to see what I am trying to accomplish. 

```
      - uses: actions/checkout@v2
      . . .
      - name: Monorepo - Set working Directory
        if: env.WASMCLOUD_REPO_STYLE == 'MONO' # Run if monorepo
        run: |
          echo "WORKING_DIR = ${GITHUB_WORKSPACE}/actors/${{ env.ACTOR_NAME }}" >> $GITHUB_ENV
      - name: Multirepo - Set working Directory
        if: env.WASMCLOUD_REPO_STYLE == 'MULTI' # Run if multirepo
        run: |
          echo "WORKING_DIR = ${GITHUB_WORKSPACE}" >> $GITHUB_ENV
      - name: Build wasmcloud actor
        run: make
        working-directory: ${{ env.WORKING_DIR }} # If I Hardcode path here it works
```
 The environment variable is showing the correct path during debugging which is formatted as: `/home/runner/work/REPO_NAME/REPO_NAME/actors/ACTOR_NAME`
 
 For step 3 if I type `working-directory: actors/ACTOR_NAME` it works (until a later issue :P where it again does not find a directory)
 
 This is my first few days with GitHub Actions so help would be really appreciated. I am not sure why context is not being accepted but a static path is. Examples often show working-directory using context.

**Comment by andar1an** [comment, marked as answer]
Needed to remove spaces around = in echo, and just use $ACTOR_NAME in path

**Comment by alightmotionapkiz** [comment]
thanks I have found this solutions
[alight motion apk](https://alightmotionapkiz.com/)

**Comment by homecurtain** [comment]
thank you giving me this information
[Homecurtain](https://homecurtain.ae/)
